File: lc-files/java/lc14.java
import java.util.ArrayList;

public class lc14 {
    public static String getLongestCommonPrefix (String[] arr) {
        int pointer = 0;
        ArrayList<String> commonPrefix = new ArrayList<>();

        if(arr.length == 0) return "";
        else if(arr.length == 1) return arr[0];
        else {
            // Get shortest length
            int shortestStrLen = arr[0].length();
            for(int i = 0; i < arr.length; i++) {
                if(arr[i].length() > shortestStrLen) {
                    shortestStrLen = arr[i].length();
                }
            }

            // Find the common prefix...
            Boolean hasSearchNotEnded = true;
            while(pointer < shortestStrLen && hasSearchNotEnded) {
                char currentChar = arr[0].charAt(pointer);
                Boolean isCharInAllStrs = true;
                for(int i = 0; i< arr.length; i++) {
                    if(arr[i].charAt(pointer) != currentChar) {
                        isCharInAllStrs = false;
                    }
                }

                if(isCharInAllStrs) {
                    commonPrefix.add(String.valueOf(currentChar));
                    pointer++;
                } else {
                    hasSearchNotEnded = false;
                }
            }
        }

        return String.join("", commonPrefix);
    }
}

File: lc-files/python/lc14.py
def getLongestCommonPrefix(strArr):
    pointer = 0;
    commonPrefix = [];

    if(len(strArr) == 0):
        return "";
    elif(len(strArr) == 1):
        return strArr[0];
    else:
        # Get Shortest Length
        shortestStrLen = len(strArr[0]);
        for i in range(len(strArr)):
            if(len(strArr[i]) < shortestStrLen):
                shortestStrLen = len(strArr[i]);

        searchNotEnded = True;
        while pointer < shortestStrLen and searchNotEnded:
            currentChar = strArr[0][pointer];
            isCharInAllStrs = True;
            for i in range(len(strArr)):
                if(strArr[i][pointer] != currentChar):
                    isCharInAllStrs = False;

            if isCharInAllStrs:
                commonPrefix.insert(currentChar);
                pointer = pointer + 1;
            else:
                searchNotEnded = False;
    return commonPrefix.join("");
